alma/lib/Services/InsuranceProductService.php

```diff
@@ -253,9 +270,18 @@ public function addInsuranceProduct(
      *
      * @return bool
      */
-    public function handleAddingProductInsurance($idProduct, $insuranceContractId, $quantity, $idCustomization, $cart = null, $destroyPost = true)
+    public function handleAddingProductInsurance($cart = null)
     {
         try {
+            if (!$this->canHandleAddingProductInsurance()) {
+                return false;
+            }
+
+            $idProduct = \Tools::getValue('id_product');
+            $insuranceContractId = \Tools::getValue('alma_id_insurance_contract');
+            $quantity = \Tools::getValue('qty');
+            $idCustomization = \Tools::getValue('id_customization');
```

Comment:
add toolFactory - dependency inversion - you can inject params to no longer depend on the execution context